src/MenuFSM.cpp

```diff
@@ -0,0 +1,129 @@
+#include "MenuFSM.h"
+
+#include <iostream>
+
+std::string stateName(MenuState state) {
+    switch(state) {
+        case PAUSING: return "PAUSING"; break;
+        case EXITING: return "EXITING"; break;
+        case PLAYING: return "PLAYING"; break;
+    }
+}
+
+MenuFSM::MenuFSM(entt::entity entity)
+    : id(entity)
+{
+    currentState = MenuState::PAUSING;
+
+    add_transition_to_new_state(MenuTransition::PAUSE, MenuState::PAUSING);
+    add_transition_to_new_state(MenuTransition::EXIT, MenuState::EXITING);
+    add_transition_to_new_state(MenuTransition::PLAY, MenuState::PLAYING);
+
+    add_valid_transitions_from_state( MenuState::PLAYING,
+        containers::Set<MenuTransition> { MenuTransition::PAUSE });
+    add_valid_transitions_from_state( MenuState::PAUSING,
+        containers::Set<MenuTransition> { MenuTransition::PLAY, MenuTransition::EXIT });
+}
+
+void MenuFSM::on_exit(MenuState fromState)
+{
+    std::cout << "exiting state: [" << stateName(fromState) << "], ";
+}
+
+/*void MenuFSM::load_game()
```

Comment:
Remove commented out code if you're not going to use it. You can find it in git history or make a comment that's more documentation to record something you want to remember, rather than leaving whole chunks like this. Or leave it in but disable it somehow / not call it.